apache/superset
# feat(dashboard): Transition to Explore with React Router

### SUMMARY

Currently, when user clicks on chart title or "Edit chart" button in Dashboard view, we first make a POST request to get form_data_key and then open Explore in a new window. In order to fully benefit from Explore SPA project, we need to stay in the SPA context when transitioning from Dashboard to Explore - which means that we should open Explore in the same tab.

This PR changes the default behaviour to open Explore in the same tab using the React Router. If user wants to open Explore in separate tab, they can do cmd+click (on Mac) or ctrl+click (on Linux/Windows). The tooltip was adjusted to make the feature easier to discover. 

Why can't we just make the chart title a link and let user "right click -> open in a new tab"? When user clicks "Edit chart", we first need to send an API request to get the form_data_key, which is required to correctly pass dashboard context to Explore.   For this reason we can't use standard HTML <a> element and instead we use a button with `onClick` callback.
In the future, we'll work on improving this behaviour, so that we can use a link to transition to Explore.

Since this change in flow (open Explore in the same tab by default) might be disruptive, user can bring back the old behaviour by enabling the new feature flag `DASHBOARD_EDIT_CHART_IN_NEW_TAB`.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/15073128/177313748-4b63608e-d761-4ef1-904c-bb4733d73648.mov


### TESTING INSTRUCTIONS

1. Go to dashboard
2. Click on chart title or "Edit chart" in menu or "Edit chart" in "View as table" modal
3. Verify that Explore opens in the same tab
4. Go back to Dashboard
5. Click on chart title while holding CMD
6. Verify that Explore opens in new tab

### ADDITIONAL INFORMATION
